### PR TITLE
[ci/train] Add Ray Train storage refactor CI tests

### PR DESCRIPTION
**Pull request by krfricke**
## Why are these changes needed?

This PR adds CI runners for the Ray Train and Tune tests with the new storage context path enabled.

Many tests are excluded at first. We will iteratively work on enabling them to avoid having to fix a bunch of issues in one giant PR.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: .buildkite/pipeline.ml.yml [hunk 1]
```diff
@@ -335,6 +337,71 @@
     - bazel test --config=ci $(./ci/run/bazel_export_options) --build_tests_only python/ray/tests/horovod/...
 
 
+
+##### STORAGE REFACTOR
+
+- label: ":steam_locomotive: Train tests and examples"
```
Comment (justinvyu) [inline .buildkite/pipeline.ml.yml:343]: I think we also need to add:

* :book: Doc tests and examples (excluding Ray AIR examples)
* :airplane: AIR tests (ray/air)
* :airplane: AIR/ML release smoke tests

GPU:
* :tv: :airplane: AIR GPU tests (ray/air)
* :tv: :steam_locomotive: Train GPU tests

Doctest would also probably fail.

Comment (krfricke) [reply .buildkite/pipeline.ml.yml:343]: Ok sounds good - I'll first take a stab with the current selection but can update later/soon.

Comment (justinvyu) [reply .buildkite/pipeline.ml.yml:343]: Got it, should we coordinate with a spreadsheet and shard the fixes?

Comment (justinvyu) [reply .buildkite/pipeline.ml.yml:343]: Also, let's update the names of these Train / Train + Tune ones to also be prefixed by "New persistence mode:"

**Review by justinvyu — APPROVED** [review]
Thanks!

**Pull request merged**